packages/plugin-image/web/src/toolbar/image-settings.js

```diff
@@ -19,29 +20,66 @@ class ImageSettings extends Component {
     this.initialState = { ...this.state };
     const { t, theme } = props;
     this.styles = mergeStyles({ styles, theme });
-
     this.updateLabel = t('ImageSettings_Update');
     this.headerText = t('ImageSettings_Header');
     this.captionLabel = t('ImageSettings_Caption_Label');
     this.captionInputPlaceholder = t('ImageSettings_Caption_Input_Placeholder');
     this.altLabel = t('ImageSettings_Alt_Label');
     this.altTooltip = 'ImageSettings_Alt_Label_Tooltip';
     this.altInputPlaceholder = t('ImageSettings_Alt_Input_Placeholder');
-    this.linkRedirectTextLabel = t('ImageSettings_Link_Label');
-    this.linkRedirectText = t('ImageSettings_Link_RedirectToToolbar');
   }
 
   propsToState(props) {
-    const {
-      componentData: { src, metadata, error },
-    } = props;
+    const { componentData } = props;
+    const { src, metadata, error, config } = componentData;
+    const isExpandEnabled = !config?.disableExpand;
+    const isRightClickEnabled = !componentData.disableRightClick;
+
     return {
       src,
       metadata,
       error,
+      isExpandEnabled,
+      isRightClickEnabled,
     };
   }
 
+  toggleState = key => () => {
+    this.setState(prevState => ({
+      [key]: !prevState[key],
+    }));
+  };
+
+  renderToggle = ({ toggleKey, labelKey, dataHook, tooltipText }) => {
+    return (
+      <div key={toggleKey} className={this.styles.imageSettings_toggleContainer}>
+        <LabeledToggle
+          style={{ paddingTop: 24 }}
+          theme={this.props.theme}
+          checked={this.state[toggleKey]}
+          label={this.props.t(labelKey)}
+          onChange={this.toggleState(toggleKey)}
+          dataHook={dataHook}
+          tooltipText={tooltipText ? tooltipText : null}
```

Comment:
This should probably work
```suggestion
          tooltipText={tooltipText}
```